src/main/java/ExceptionHandler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,27 @@
+package ExceptionHandler;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+@ControllerAdvice
+public class GlobalExceptionHandler {
+    @ExceptionHandler({UserNotFoundException.class})
+    public ResponseEntity<Object> handleUserNotFoundException(UserNotFoundException exception) {
+        return ResponseEntity
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
```

Comment:
server error is not a good response if the user is not found. With server error you are implying that something wrong happened in the backend, but nothing did, you just can't find the user